app/scripts/main.js

```diff
@@ -658,13 +658,13 @@ window.iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
 
       if (_self.cohorts_.length === 1) { // to query single study
         if (QueryByGeneTextArea.isEmpty()) {
-          QueryByGeneUtil.toMainPage(_self.cohorts_[0], _self.stat().studies);
+          QueryByGeneUtil.toMainPage(_self.cohorts_[0], vm_.hasfilters ? _self.stat().studies : undefined);
```

Comment:
Question, should `cancer_study_list` be input studies or selected studies after applying filters?

Reply:
selected studies after applying filters

Reply:
Sorry, I got this. If no filters applied first parameter would be used else studies from stat would be used.

Reply:
instead of passing `undefined` and checking it, why not just pass `_self.stat().studies` and `hasfilters` flag.